tests/ref/test_schema.py

```diff
@@ -36,3 +50,66 @@ def test_create_schema_from_records(
 
     for segment in auto_schema.segments:
         assert type(segment.id) is UUID
+
+
+class TestSegmentNameParser:
+    @pytest.mark.parametrize(
+        "expected_result, test_strings",
+        [
+            (SegmentName(prefix="DNA", key="A"), ["DNA A", "DNA_A", "DNA-A"]),
+            (SegmentName(prefix="RNA", key="BN"), ["RNA BN", "RNA_BN", "RNA-BN"]),
+            (SegmentName(prefix="DNA", key="U3"), ["DNA U3", "DNA U3", "DNA-U3"]),
+        ],
+    )
+    def test_ok(self, expected_result: str, test_strings: list[str]):
+        assert (
+            parse_segment_name(test_strings[0])
+            ==
+            parse_segment_name(test_strings[1])
+            ==
+            parse_segment_name(test_strings[2])
+            ==
+            expected_result
+        )
+
+    @pytest.mark.parametrize(
+        "fail_case",
+        ["", "*V/", "51f9a0bc-7b3b-434f-bf4c-f7abaa015b8d"]
+    )
+    def test_fail(self, fail_case: str):
+        with pytest.raises(ValueError):
+            parse_segment_name(fail_case)
+
+    @pytest.mark.parametrize(
+        "expected_result, test_name, test_moltype",
+        [
+            (SegmentName(prefix="DNA", key="A"), "A", MolType.DNA),
+            (SegmentName(prefix="RNA", key="BN"), "BN", MolType.RNA),
+            (SegmentName(prefix="DNA", key="U3"), "U3", MolType.DNA),
+        ],
+    )
+    def test_parse_from_record_ok(self, expected_result, test_name, test_moltype, mocker):
+        mock_record = MockNCBIGenbank(moltype=test_moltype, source=MockNCBISource(test_name))
+
+        mocker.patch.object(NCBIGenbank, "__new__", return_value=mock_record)
+
+        record = NCBIGenbank()
```

Comment:
Sorry, you need to use a real `NCBIGenbank` object. This current solution is dark magic.

Reply:
But my voodoo doll 🥲

I was hoping not to have to use `NCBIClient` in test_schema. Perhaps it is time to muck with PolyFactory again.